pkg/metrics/metrics.go

```diff
@@ -165,10 +165,34 @@ var (
 			ProvisionerLabel,
 		},
 	)
+	MachinesDisruptedCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: Namespace,
+			Subsystem: machineSubsystem,
+			Name:      "disrupted",
+			Help:      "Number of machines disrupted in total by Karpenter. Labeled by reason the machine was disrupted.",
+		},
+		[]string{
+			ReasonLabel,
+			ProvisionerLabel,
+		},
+	)
+	MachinesDriftedReasonCounter = prometheus.NewCounterVec(
```

Comment:
```suggestion
	MachinesDriftedCounter = prometheus.NewCounterVec(
```

Reply:
Probably better to just have a separate counter that's called `karpenter_machines_drifted` and contains the reason